src/lib/device-online-state.ts

```diff
@@ -338,7 +345,7 @@ class DeviceOnlineStateManager {
 						uuid,
 						DeviceOnlineStates.Online,
 						DeviceOnlineStates.Timeout,
-						timeoutSeconds,
+						Math.ceil(timeoutSeconds), // always make this a whole number of seconds, and round up to make sure we dont expire too soon...
```

Comment:
Redis keys can only expire in whole seconds, so we round up to make sure we don't expire too soon.